industrial_ci/ci_main.sh

```diff
@@ -65,18 +65,30 @@ if [[ "$ROS_DISTRO" == "kinetic" ]] && ! [ "$IN_DOCKER" ]; then
   travis_time_end  # build_docker_image
 
   travis_time_start run_travissh_docker
+
+  #forward ssh agent into docker container
+  if [ "$SSH_AUTH_SOCK" ]; then
+      export SSH_DOCKER_CMD="-t -i -v $(dirname $SSH_AUTH_SOCK):$(dirname $SSH_AUTH_SOCK) -e SSH_AUTH_SOCK=$SSH_AUTH_SOCK"
+  else
+      export SSH_DOCKER_CMD=""
+  fi
+
   docker_target_repo_path=/root/ci_src
   docker_ici_pkg_path=${ICI_PKG_PATH/$TARGET_REPO_PATH/$docker_target_repo_path}
   docker run \
       --env-file ${ICI_PKG_PATH}/docker.env \
       -e TARGET_REPO_PATH=$docker_target_repo_path \
+      $SSH_DOCKER_CMD \
       -v $TARGET_REPO_PATH/:$docker_target_repo_path industrial-ci/xenial \
       /bin/bash -c "cd $docker_ici_pkg_path; source ./ci_main.sh;"
   retval=$?
   if [ $retval -eq 0 ]; then HIT_ENDOFSCRIPT=true; success 0; else exit; fi  # Call  travis_time_end  run_travissh_docker
 fi
 
 travis_time_start init_travis_environment
+#setup github.com ssh key
+mkdir -p ~/.ssh
+ssh-keyscan github.com >> ~/.ssh/known_hosts
```

Comment:
Ok, confirmed that the key is not printed (I should've done that first).

> ```
> ssh-keyscan github.com >> ~/.ssh/known_hosts 2> /dev/null
> ```

+1
I'm not a security expert so just prefer sticking to not disclosing network-related info as much as possible.